### PR TITLE
Ft update profile UqDcj25

### PR DESCRIPTION
**Pull request by 41hulk**
#### What does this PR do?
This PR help to **UPDATE** user to route `/api/users/update`
#### Description of Task to be completed?
The operator and driver should be able to update information about their own profile.
The operator and driver profile should be able accessible publicly.
Appropriate error messages are displayed on unsuccessful attempts.
```PATCH  /api/users/update```
#### How should this be manually tested?
**Testing** `/api/users/update` 

- Make a `.env` file and fill it with information as provided in the `.ev_example` file.
- First get a token granted from POST signin request on route `/api/users/signin` which is token you will be using to make requests along with the header `Authorization : token` on all routes because all routes are protected to only administrator and operators and also a body with email and password credentials of a sample admin from the mocks folder.

**Testing using postman**
- Make a PATCH request to route `/api/users/update/{userId}`  along with data that involves fields and their values to **update** the specific users with his/her ID

#### What are the relevant trello board?
Delivers UqDcj25i

### CODE REVIEW

**Review by KundwaBruno — APPROVED** [review]

**Review by KundwaBruno — COMMENTED** [review]
Well done!

**Review by GuevaraMarius — APPROVED** [review]

File: src/controllers/usersController.js [hunk 1]
```diff
@@ -155,3 +155,18 @@ export const logout = async (req, res) => {
...
+    try {
+      const userId = req.params.id;
+
+      const updatedUser = User.update({...req.body},
+        { where: { id: userId } })
+
+      successRes(res,200,res.__('User Updated'), updatedUser);
+      
+    } catch (error){
+        console.log(error);
```
Comment (raymond42) [inline src/controllers/usersController.js]: can you remove this console.log()?

File: src/controllers/usersController.js [hunk 2]
```diff
@@ -155,3 +155,18 @@ export const logout = async (req, res) => {
     return errorRes(res, 500, res.__('There was error loging out'));
   }
 };
+
+export const updateProfile = async (req,res) => {
+    try {
+      const userId = req.params.id;
+
+      const updatedUser = User.update({...req.body},
```
Comment (raymond42) [inline src/controllers/usersController.js]: let's have await before User. And why are we having spread operator on req.body?

File: src/controllers/usersController.js [hunk 3]
```diff
@@ -155,3 +155,18 @@ export const logout = async (req, res) => {
...
+        { where: { id: userId } })
+
+      successRes(res,200,res.__('User Updated'), updatedUser);
+      
+    } catch (error){
+        console.log(error);
+        errorRes(res, 500, res.__("There was an error while updating the User"));
+      }
+};
```
Comment (raymond42) [inline src/controllers/usersController.js]: can you add an empty line at the end? check if your eslint is working

File: src/test/mocks/mockUsers.js [hunk 4]
```diff
@@ -19,3 +19,62 @@ export const wrongPwd = {
...
+  role: 'driver',
+};
+
+
+export const noNameFound={
+  "email": "Andy@gmail.com",
+  "role" : "driver",
+  "phone" : "0785464542",
+  "language": "en",
+  "nationalId":"1392417023348424"
```
Comment (raymond42) [inline src/test/mocks/mockUsers.js]: should the nationalId not be an integer? check on that

File: src/test/users.test.js [hunk 5]
```diff
@@ -140,4 +139,88 @@ describe('Users Related Tests', async () => {
...
+      .set('auth', token);
+
+    expect(res.status).to.be.equal(200);
+    expect(res.body).to.have.property(
+      'message',
+      'User Updated',
+    );
+  });
+
+  it('should fiind Names not provided', async() => {
```
Comment (raymond42) [inline src/test/users.test.js]: let's have a good description of the tests. For example on this one you can say, it should throw an error when the names are not provided

**Review by raymond42 — CHANGES_REQUESTED** [review]
please attend to the comments and revert to me

**Review by mansurissa — APPROVED** [review]
Great job man

**Review by raymond42 — APPROVED** [review]
Nice work

**Pull request merged**